File: level2/큰수만들기.py
def solution(number, k):
    i = 0
    while k:
        k -= 1
        for _ in range(i,len(number)):
            if i+1 == len(number):
                number = number[:-1]
                break
                
            if int(number[i]) < int(number[i+1]):
                number = number[:i]+number[i+1:]
                i = max(i-1,0)
                break
            i += 1
        
    return number

File: level2/큰수만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    int len = number.size(), start = 0;
    int end = len-k;
    
    string answer = "";
    while(end--){
        int tmp_num = -1;
        for(int i = start; i < len-end; i++){
            if(tmp_num < (int)(number.at(i)-'0')){
                start = i+1;
                tmp_num = (int)(number.at(i)-'0');
            }
        }
        answer += tmp_num + '0';
    }
    
    return answer;
}
